### PR TITLE
[LLVM] Implement SIMDArray 

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**

### CODE REVIEW

**Comment by Thirumalai-Shaktivel** [comment]
Ready for review!

**Comment by Thirumalai-Shaktivel** [comment]
LLVM:
```console
lfortran integration_tests/matmul_01.f90 
Size (n x n): n = 96
Iter = 1
Size MB: 3.51562500000000000e-02
Fortran intrinsic matmul:
Time:  4.05299999999999931e-03
Clock cycles per element:
Theoretical performance peak: 6.25000000000000000e-02 cycles
Measured:                     1.46592881944444429e+01 cycles
Percent peak:                 4.26350851221317551e-01 %

matmul2:
Error: 0.00000000e+00
Time:  2.45000000000000079e-03
Clock cycles per element:
Theoretical performance peak: 6.25000000000000000e-02 cycles
Measured:                     8.86140046296296546e+00 cycles
Percent peak:                 7.05306122448979389e-01 %

matmul1:
Error: 0.00000000e+00
Time:  2.48899999999999989e-03
Clock cycles per element:
Theoretical performance peak: 6.25000000000000000e-02 cycles
Measured:                     9.00245949074074048e+00 cycles
Percent peak:                 6.94254720771394163e-01 %
```

File: integration_tests/simd_02.f90 [hunk 1]
```diff
@@ -1,14 +1,14 @@
 program simd_02
     !LF$ attributes simd :: A, B, res
     real :: A(4), B(4), C(8), res(4)
-    C = 43
+    C = [3.14, 7.25, 0.98, 5.67, 2.35, 9.42, 6.11, 1.99]
```
Comment (certik) [inline integration_tests/simd_02.f90:4]: We still need to test `C = 43` somewhere, but we can do that later.

**Review by certik — APPROVED** [review]
I think this looks great to get started, so I am merging this. Thanks!

File: integration_tests/simd_02.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/simd_02.f90:4]: It seems to be tested in `simd_01`, if you want any tests to be added, do let me know.

Comment (certik) [reply integration_tests/simd_02.f90:4]: Ok, that's fine. Good for now.

**Pull request merged**